<?xml version="1.0" ?>
<launch xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="https://gist.githubusercontent.com/nalt/dfa2abc9d2e3ae4feb82ca5608090387/raw/roslaunch.xsd">
    <!-- Arguments -->
    <arg name="log_level" default="INFO" />
    <arg name="server_domain" default="0.0.0.0"/>
    <arg name="server_port" default="5000"/>
    <arg name="api_prefix" default="/api/v1/robot" />
    <arg name="hardware_version" default="ned2" />
    
    <node name="niryo_robot_system_api_client" pkg="niryo_robot_system_api_client" type="system_api_client_node.py" output="screen">
        <rosparam file="$(find niryo_robot_system_api_client)/config/$(arg hardware_version)/default.yaml" command="load"/>
        <param name="log_level" type="string" value="$(arg log_level)" />
        <param name="server_domain" type="string" value="$(arg server_domain)"/>
        <param name="server_port" type="string" value="$(arg server_port)"/>
        <param name="api_prefix" type="string" value="$(arg api_prefix)" />
    </node>
</launch>
